cpp/CMakeLists.txt

```diff
@@ -37,7 +37,7 @@ else()
 endif()
 
 # Check for OPENTELEMETRY-CPP for distributed tracing
-find_package(opentelemetry-cpp)
+find_package(opentelemetry-cpp CONFIG REQUIRED)
```

Comment:
Is it going to be `REQUIRED`? I am all for it (less possible build configurations), but I imagine some users might want to build Proton without opentelemtry cpp.

Reply:
I would doubt it generally, but especially since its not yet even in a released state. I had suggested on IRC shortly before your comment the next step would be to make it optional.

Reply:
should be enough to delete the `REQUIRED`. There already is the `cmake_dependent_option` thing for this, for dealing with the optionality below in the file.

Reply:
Yep, I noted that it mostly seemed optional already apart from the find.

Reply:
Done! I've deleted the `REQUIRED`. 
But this time I've changed the `cmake_dependent_option` to `option`. `option` will also work right?

Reply:
I'll need to defer to the folks with cmake experience at this point to suggest on the specific approach, I dont actually know myself.